titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/connector/NodePredicates.java

```diff
@@ -47,4 +51,24 @@ static boolean isOwnedByScheduler(String schedulerName, V1Node node) {
                 KubeConstants.TAINT_SCHEDULER.equals(taint.getKey()) && schedulerName.equals(taint.getValue())
         );
     }
+
+    @VisibleForTesting
+    static boolean hasBadCondition(V1Node node, Function<String, Matcher> badConditionExpression,
+                                   int nodeConditionTransitionTimeThresholdSeconds) {
+        if (node.getStatus() != null && node.getStatus().getConditions() != null) {
+            return node.getStatus().getConditions().stream()
+                    .anyMatch(v1NodeCondition -> badConditionExpression.apply(v1NodeCondition.getType()).matches() &&
+                            v1NodeCondition.getStatus().equals("True") &&
```

Comment:
Boolean.valueOf to make this cleaner/safer.

Reply:
Good catch